akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1473,6 +1471,8 @@ private[akka] abstract class ReadWriteAggregator extends Actor {
 
   def timeout: FiniteDuration
   def nodes: Set[Address]
+  def unreachable: Set[Address]
+  def reachableNodes: Set[Address] = nodes -- unreachable
```

Comment:
The advice is based on a presentation I watched some years ago. Can't find it now. I think it is because Set specific operations are overridden and optimized in the specific Set implementations, see `HashSet.diff`

Same with `union` and some others.